control_protocol.md

```diff
@@ -330,17 +330,490 @@ A Component MUST also offer a list of all possibly callable methods in accordanc
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "sign_out",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "coordinator_sign_in",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "coordinator_sign_out",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "set_nodes",
+            "params": [
+                {
+                    "name": "nodes",
+                    "schema": {
+                        "type": "object",
+                        "additionalProperties": true
+                    },
+                    "required": true
+                }
+            ],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "set_remote_components",
+            "params": [
+                {
+                    "name": "components",
+                    "schema": {
+                        "type": "array",
+                        "items": {
+                            "type": "string"
+                        }
+                    },
+                    "required": true
+                }
+            ],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "compose_global_directory",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "object",
+                    "additionalProperties": true
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "compose_local_directory",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "object",
+                    "additionalProperties": true
+                },
+                "required": true
+            },
+            "description": "Compose a dictionary with the local directory."
```

Comment:
What are you trying to say with "Compose" here?

Reply:
"Create": It composes the full dictionary from the different sources.

Reply:
Maybe "Return the local directory as a dictionary" and name `return_local_directory`?
If we encode the return type in the schema, we might leave that part out of the docstring (a dict is not called a "dictionary" in all languages).

Reply:
We have to see what to do with dictionaries, as json schema does not seem to support them (or I do not see it).

Reply:
I renamed the methods (see general discussion), such that this conversation can be resolved, I think.

Reply:
thanks!